File: problem-3.py
from math import sqrt


def isDivisable(num1, num2):
    result = num1 % num2 == 0
    return result

def isPrime(num):
    if num >= 1 and num <= 3:
        return True
    return sqrt(num) * sqrt(num) == num

def main():
    number = 600851475143
    factor = 1

    for count in range(3, int(sqrt(number))):
        if isDivisable(number, count) and isPrime(count):
            factor = count

    return factor

print(main())

File: ruby/problem-3.rb
class Challenge
  def isDivisable(num1, num2)
    return num1 % num2 == 0
  end

  def isPrime(num)
    if num >= 1 && num <= 3 then
      return TRUE
    end
    return Math.sqrt(num) * Math.sqrt(num) == num
  end

  def main(value)
    primes = []
    for num in (3...Math.sqrt(value).round()).to_a
      if self.isDivisable(value, num) && self.isPrime(num) then
        primes.push(num)
      end
    end
    return primes
  end
end

c = Challenge.new
puts c.main(600851475143)
# puts c.isPrime(13)
